core/templates/dev/head/pages/collection_editor/CollectionEditorNavbarDirective.js

```diff
@@ -68,11 +72,11 @@ oppia.directive('collectionEditorNavbar', [function() {
           // action since it is not reversible.
           CollectionRightsBackendApiService.setCollectionPublic(
             $scope.collectionId, $scope.collection.getVersion()).then(
-            function() {
+            function(response) {
               // TODO(bhenning): There should be a scope-level rights object
               // used, instead. The rights object should be loaded with the
               // collection.
-              $scope.isPrivate = false;
+              $scope.isPrivate = response.is_private;
```

Comment:
For safety, maybe $log.error() if this is not false, where the error message is something like 'Backend indicated a collection was successfully published, but response.data.is_private returned true'? Hopefully this will never be triggered, but if it is, we'll know where to look.